### PR TITLE
option to select between different optimizers added

### PR DESCRIPTION
**Pull request by allaffa**

### CODE REVIEW

File: hydragnn/utils/optimizer.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+
+def selected_optimizer(model, config):
+    optimizer = None
+
+    if config["optimizer"] == "SGD":
+        optimizer = torch.optim.SGD(model.parameters(), lr=config["learning_rate"])
+    elif config["optimizer"] == "Adam":
+        optimizer = torch.optim.Adam(model.parameters(), lr=config["learning_rate"])
+    elif config["optimizer"] == "Adadelta":
```
Comment (streeve) [inline hydragnn/utils/optimizer.py:11]: typo

File: hydragnn/utils/optimizer.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        optimizer = torch.optim.Adamdelta(
+            model.parameters(), lr=config["learning_rate"]
+        )
+    elif config["optimizer"] == "Adagrad":
```
Comment (streeve) [inline hydragnn/utils/optimizer.py:15]: typo

**Review by streeve — CHANGES_REQUESTED** [review]
Like Pei mentioned before, adding AdamW as default to keep backwards compatibility will be good (in `update_config` I think?)

File: hydragnn/utils/optimizer.py (hunk 1 above)
Comment (allaffa) [reply hydragnn/utils/optimizer.py:11]: Thanks.

File: hydragnn/utils/optimizer.py (hunk 2 above)
Comment (allaffa) [reply hydragnn/utils/optimizer.py:13]: Thanks.

**Comment by allaffa** [comment]
> Like Pei mentioned before, adding AdamW as default to keep backwards compatibility will be good (in `update_config` I think?)

`run_training.py` calls `update_config`
If I pass "AdamW" into `update_config`, this would imply that the optimizer chosen gets overwritten every time `run_training.py` is called. Correct?

**Comment by allaffa** [comment]
> Like Pei mentioned before, adding AdamW as default to keep backwards compatibility will be good (in `update_config` I think?)
@streeve 
Valid point. Done.

File: hydragnn/utils/optimizer.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import torch
+
+
+def selected_optimizer(model, config):
```
Comment (streeve) [inline hydragnn/utils/optimizer.py:4]: How about `select_optimizer` or `choose_optimizer`

Comment (allaffa) [reply hydragnn/utils/optimizer.py]: I changed the name to `select_optimizer`

**Review by allaffa — COMMENTED** [review]
I changed the name of the utility as suggested by @streeve

**Review by streeve — APPROVED** [review]

**Pull request merged**